go.mod

```diff
@@ -3,7 +3,7 @@ module github.com/cosmos/iavl
 go 1.21
 
 require (
-	cosmossdk.io/log v1.3.1
+	cosmossdk.io/core v0.12.1-0.20240514205955-97c9bbb0341b
```

Comment:
This does enshrine protobuf a bit more (dependency of core) https://github.com/cosmos/cosmos-sdk/blob/dac569c3a2a5277cec1b5ecba978074ab45def80/core/go.mod#L8-L9.
But it looks like that's already a part of iavl anyway.

Reply:
was thinking the same when I was working on it and tried to remove it but seems like a large battle for not much benefit